### PR TITLE
Require a known good version of .NET SDK be installed.

### PR DESCRIPTION
**Pull request by JoeRobich**
Since we fail when loading SDKs with mismatched NuGet.Framework, we can limit our SDK discovery to a known good SDK version.

### CODE REVIEW

File: src/OmniSharp.Host/HostHelpers.cs [hunk 1]
```diff
@@ -28,6 +29,11 @@ public static int Start(Func<int> action)
 
                 return action();
             }
+            catch (MSBuildNotFoundException mnfe)
+            {
+                Console.Error.WriteLine(mnfe.Message);
+                return 0xbad;
```
Comment (JoeRobich) [inline src/OmniSharp.Host/HostHelpers.cs:35]: Can make this a unique exit code.

File: src/OmniSharp.Host/MSBuild/Discovery/Extensions.cs [hunk 2]
```diff
@@ -72,7 +72,7 @@ public static void RegisterDefaultInstance(this IMSBuildLocator msbuildLocator,
             }
             else
             {
-                logger.LogError("Could not locate MSBuild instance to register with OmniSharp");
+                throw new MSBuildNotFoundException("Could not locate MSBuild instance to register with OmniSharp.");
```
Comment (JoeRobich) [inline src/OmniSharp.Host/MSBuild/Discovery/Extensions.cs:75]: Further loading isn't useful and clutters the output.

File: src/OmniSharp.Host/MSBuild/Discovery/Providers/MicrosoftBuildLocatorInstanceProvider.cs [hunk 3]
```diff
@@ -22,9 +22,17 @@ public override ImmutableArray<MSBuildInstance> GetInstances()
         {
 
 #if NETCOREAPP
+            const string DotNetSdkVersion = "6.0.100";
```
Comment (JoeRobich) [inline src/OmniSharp.Host/MSBuild/Discovery/Providers/MicrosoftBuildLocatorInstanceProvider.cs:25]: Should be easy to find and replace this version string.

**Review by filipw — APPROVED** [review]
yes I've suggested this yesterday  in a comment somewhere - thanks!

**Pull request merged**